api/envoy/config/wasm/v2/wasm.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+package envoy.config.wasm.v2;
+
+option java_outer_classname = "WasmProto";
+option java_multiple_files = true;
+option java_package = "io.envoyproxy.envoy.config.wasm.v2";
+
+import "envoy/api/v2/core/base.proto";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: Wasm service]
+
+// Configuration for a Wasm VM.
+// [#next-free-field: 6]
+message VmConfig {
+  // An ID which will be used along with a hash of the wasm code (or null_vm_id) to determine which
+  // VM will be used for the plugin. All plugins which use the same vm_id and code will use the same
+  // VM. May be left blank.
+  string vm_id = 1;
+
+  // The Wasm runtime type (see source/extensions/common/wasm/well_known_names.h).
+  string runtime = 2;
+
+  // The Wasm code that Envoy will execute.
+  api.v2.core.AsyncDataSource code = 3;
+
+  // The Wasm configuration string used on initialization of a new VM (proxy_onStart).
+  string configuration = 4;
+
+  // Allow the wasm file to include pre-compiled code on VMs which support it.
+  bool allow_precompiled = 5;
+}
+
+// Base Configuration for Wasm Plugins e.g. filters and services.
+message PluginConfig {
+  // A unique name for a filters/services in a VM for use in identifying the filter/service if
+  // multiple filters/services are handled by the same vm_id and root_id and for logging/debugging.
+  string name = 1;
```

Comment:
I think it would be interesting to lift this out to the generic Envoy filter interface. @alexburnos has pointed out that today, the filter name is a conflation of name and type. I don't think that would be a ton of work and would make WASM amenable to some of the filter chain federation work we've been discussing.

Reply:
Agreed.  How do you want to proceed with that?  Do you have a suggested location for this?

Reply:
E.g. https://github.com/envoyproxy/envoy/blob/master/api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto#L611. PRs welcome :)

Reply:
How do you want to proceed on this?  Do we want to change "name" -> "type" and then add a new "name" ?   Do we want to add an "id" which is the non-type name?  Do we want to do this 2 step, where we add  "Id" and "type" which is a duplicate of "name" (i.e. reads from "name" if "type" isn't set) then deprecate "name" over time?

Let's discuss this on an issue:  https://github.com/envoyproxy/envoy/issues/9466

Reply:
https://github.com/envoyproxy/envoy/pull/9391 is @kyessenov 's PR for this issue.